src/vrf03.rs

```diff
@@ -550,4 +550,52 @@ mod test {
             assert_eq!(output[..], hex::decode(vector[3]).unwrap());
         }
     }
+
+    use std::{fs::File, path::Path};
+
+    #[test]
+    fn check_against_cardano_base() {
+        check_against_golden("./tests/test_vectors/vrf_ver03_generated_1");
+        check_against_golden("./tests/test_vectors/vrf_ver03_generated_2");
+        check_against_golden("./tests/test_vectors/vrf_ver03_generated_3");
+        check_against_golden("./tests/test_vectors/vrf_ver03_generated_4");
+        check_against_golden("./tests/test_vectors/vrf_ver03_standard_10");
+        check_against_golden("./tests/test_vectors/vrf_ver03_standard_11");
+        check_against_golden("./tests/test_vectors/vrf_ver03_standard_12");
+    }
+
+    fn check_against_golden(file_path: &str) {
+        let json_file_path = Path::new(file_path);
+        let file = File::open(json_file_path).unwrap();
+        let test_vector: serde_json::Value =
+            serde_json::from_reader(file).expect("JSON was not well-formatted");
+
+        let vrf_name = test_vector.get("vrf").unwrap().as_str().unwrap();
```

Comment:
We should create a `VRFTestVector` structure and have serve take care of deserialisation, along with proper types for each member. Using automatic derivation should remove quite a lot of boilerplate conversion I think

Reply:
here is some code I wrote to handle deserialisation from hex encoded string into bytes: https://github.com/pragma-org/amaru/blob/81a72af8a5b145ecfcee9a255d129fb703efdccf/crates/ouroboros/tests/validation.rs#L109

Reply:
thank you very much for this remark and request. Used serde and its deserialization as requested!